src/tasker/tasker.module.ts

```diff
@@ -1,10 +1,55 @@
-import { Module } from '@nestjs/common'
-import { ImporterTaskerService } from './importer-tasker/importer-tasker.service'
-import { MailerTaskerService } from './mailer-tasker/mailer-tasker.service'
-import { PaymentDebitorTaskerService } from './payment-debitor-tasker/payment-debitor-tasker.service'
-import { WebhookTaskerService } from './webhook-tasker/webhook-tasker.service'
+import { cwd } from 'process';
+import { DynamicModule, Module, Type } from '@nestjs/common';
+import { ConfigModule, ConfigService } from '@nestjs/config';
+import { TypeOrmModule } from '@nestjs/typeorm';
 
-@Module({
-  providers: [MailerTaskerService, PaymentDebitorTaskerService, WebhookTaskerService, ImporterTaskerService],
-})
-export class TaskerModule {}
+import { AppDataSourceConfig } from '~/data-source';
+import { ExampleTasker } from './example.tasker';
+import { BullModule } from '@nestjs/bull';
+import { TaskerType } from './tasker';
+
+@Module({})
+export class TaskerModule {
+  static createTaskerProvider(workerName: TaskerType): Type<any> {
+    switch (workerName) {
+      case TaskerType.EXAMPLE_TASKER:
+        return ExampleTasker;
+      default:
+        throw new Error(`Given tasker name is not exists: ${workerName}`);
+    }
+  }
+
+  static forRoot(options: { workerName: TaskerType, nodeEnv: string }): DynamicModule {
+    const { workerName, nodeEnv } = options;
+    return {
+      module: TaskerModule,
+      imports: [
+        TypeOrmModule.forRoot(AppDataSourceConfig),
+        ConfigModule.forRoot({
+          envFilePath: `${cwd()}/.env${nodeEnv ? `.${nodeEnv}` : ''}`,
+          isGlobal: true,
+        }),
+        BullModule.forRootAsync({
+          imports: [ConfigModule],
+          useFactory: (configService: ConfigService<{
+            QUEUE_REDIS_URI: string;
+          }>) => {
+            const queueRedisUri = configService.getOrThrow('QUEUE_REDIS_URI');
```

Comment:
the same as the above. it is better to use constant string variable

Reply:
As previous comment.